### PR TITLE
Organize Github Actions into separate jobs to add parallelization

### PR DESCRIPTION
**Pull request by antekresic**

### CODE REVIEW

File: .github/workflows/go-scheduled.yml [hunk 1]
```diff
@@ -31,17 +39,50 @@ jobs:
...
+        key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+        restore-keys: |
+          ${{ runner.os }}-go-
+
+    - name: Get dependencies
+      run: |
+        go get -v -t -d ./...
+        if [ -f Gopkg.toml ]; then
+            curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
+            dep ensure
```
Comment (blagojts) [inline .github/workflows/go-scheduled.yml:81]: since we don't use dep, can we remove this?

File: .github/workflows/go-scheduled.yml [hunk 2]
```diff
@@ -50,3 +91,49 @@ jobs:
       run: |
         go generate ./...
         git diff --exit-code
+
+  test-end-to-end:
+    name: Test end-to-end extension combinations
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
```
Comment (blagojts) [inline .github/workflows/go-scheduled.yml:99]: 🔥

File: .github/workflows/go.yml [hunk 3]
```diff
@@ -22,6 +22,14 @@ jobs:
     - name: Check out code into the Go module directory
```
Comment (blagojts) [inline .github/workflows/go.yml:22]: what's the diff between this and go-scheduled.yml?

**Review by blagojts — APPROVED** [review]
Looks great!

I left a question, for my own info, in the review. 
PS: Should we update the scripts to use go 1.15?

File: .github/workflows/go.yml (hunk 3 above)
Comment (antekresic) [reply .github/workflows/go.yml:22]: Scheduled tests run once a day. This runs on PRs and code pushes to master branch.

Comment (antekresic) [reply .github/workflows/go.yml:22]: Scheduled tests also run on extended dataset.

File: .github/workflows/go-scheduled.yml (hunk 1 above)
Comment (antekresic) [reply .github/workflows/go-scheduled.yml:81]: I guess we can, I didn't want to touch it for this PR.

**Comment by VineethReddy02** [comment]
LGTM! Nice that you added `fmt`, `lint` into go-scheduled.yml

**Review by VineethReddy02 — APPROVED** [review]

**Pull request merged**